src/main/java/org/folio/processing/mapping/defaultmapper/processor/Processor.java

```diff
@@ -145,6 +160,51 @@ private boolean canProcessSubFieldMapping(JsonObject subFieldMapping, DataField
     return true;
   }
 
+  private Map<Integer, JsonObject> getFieldMappingIndicators(JsonArray mappingEntry) {
+    Map<Integer, JsonObject> indicatorsMap = new HashMap<>();
+    for (int i = 0; i < mappingEntry.size(); i++) {
+      JsonObject indicatorsSet = null;
+      if(mappingEntry.getJsonObject(i).containsKey("entity")){
+        JsonObject subFieldEntityMapping = (JsonObject) ((JsonArray) (mappingEntry.getJsonObject(i)
+          .getValue("entity"))).getValue(0);
+        if (subFieldEntityMapping.containsKey("indicators")) {
+          indicatorsSet = (JsonObject) subFieldEntityMapping.getValue("indicators");
+        }
+      }
+      indicatorsMap.put(i, indicatorsSet);
+    }
+    return indicatorsMap;
+  }
+
+  private boolean checkOnIndicatorsCorrespondence(JsonObject subFieldMappingEntity, String dataFieldInd1, String dataFieldInd2) {
+    JsonObject subFieldMappingIndicators = (JsonObject) subFieldMappingEntity.getValue("indicators");
+    String subFieldMappingInd1 = subFieldMappingIndicators.getString("ind1");
+    String subFieldMappingInd2 = subFieldMappingIndicators.getString("ind2");
+
+    return (Objects.nonNull(subFieldMappingInd1) && Objects.nonNull(subFieldMappingInd2)
+      && subFieldMappingInd1.equals(dataFieldInd1) && subFieldMappingInd2.equals(dataFieldInd2))
+      || (subFieldMappingInd1.equals(dataFieldInd1) && subFieldMappingInd2.equals("*"))
+      || (subFieldMappingInd2.equals(dataFieldInd2) && subFieldMappingInd1.equals("*"));
```

Comment:
```suggestion
    return (dataFieldInd1.equals(subFieldMappingInd1) || "*".equals(subFieldMappingInd1))
        && (dataFieldInd2.equals(subFieldMappingInd2) || "*".equals(subFieldMappingInd2));
```
propose to refactor condition based on mutual exclusion, "*" we can extract to constant as WILDCARD_INDICATOR

Reply:
Thanks Ruslan, it is done!